nixos/modules/services/matrix/pantalaimon-options.nix

```diff
@@ -0,0 +1,118 @@
+{ config, lib, name, ... }:
+
+with lib;
+{
+  options = {
+    headless = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        <literal>pantalaimon</literal> supports two modes of operation:
+
+        <itemizedlist>
+          <listitem><para>a "headless" mode, intended for unsupervised services (for example bots like <literal>mjolnir</literal>)</para></listitem>
+          <listitem><para>a "non-headless" mode, intended as a user-service (for example when using a matrix client which does not support E2E like <literal>fractal</literal>)</para></listitem>
+        </itemizedlist>
+      '';
+    };
+
+    dataPath = mkOption {
+      type = types.path;
+      default = "/var/lib/pantalaimon-${name}";
+      description = ''
+        The directory where <literal>pantalaimon</literal> should store e.g. the database file in.
+      '';
+    };
+
+    logLevel = mkOption {
+      type = types.enum [ "info" "warning" "error" "debug" ];
+      default = "warning";
+      description = ''
+        Set the log level of the daemon.
+      '';
+    };
+
+    notifications = mkOption {
+      type = types.bool;
+      default = true;
```

Comment:
If you decide to move the non-headless version to hm, this should be false by default I think.

Or this default should be dependent on the `headless` option

Reply:
Currently in progress of changing it, I've removed the `headless` option and the `notifications`, `useKeyring` and `ignoreVerifications` and set the values in the generated config in `mkConfigFile` for by-default headless operation like 

```nix
{}: recursiveUpdate {
# ...
    SSL = instanceConfig.ssl;

    # Set some settings to prevent user interaction for headless operation
    IgnoreVerification = true;
    Notifications = false;
    UseKeyring = false;
} instanceConfig.extraSettings
```

which can be overridden with the `extraSettings` attribute, if needed.